internal/pkg/agent/cmd/diagnostics.go

```diff
@@ -575,23 +581,61 @@ func zipLogs(zw *zip.Writer) error {
 			return nil
 		}
 
-		lf, err := os.Open(path)
-		if err != nil {
-			return fmt.Errorf("unable to open log file: %w", err)
+		return saveLogs(name, path, zw)
+	})
+}
+
+func collectEndpointSecurityLogs(zw *zip.Writer, specs map[string]program.Spec) error {
+	spec, ok := specs["endpoint-security"]
+	if !ok {
+		return nil
+	}
+
+	logs, ok := spec.LogPaths[runtime.GOOS]
+	if !ok {
+		return nil
+	}
+
+	logPath := filepath.Dir(logs) + string(filepath.Separator)
+	return filepath.WalkDir(logPath, func(path string, d fs.DirEntry, fErr error) error {
+		if fErr != nil {
+			if stderrors.Is(fErr, fs.ErrNotExist) {
+				return nil
+			}
+
+			return fmt.Errorf("unable to walk log dir: %w", fErr)
 		}
-		zf, err := zw.Create("logs/" + name)
-		if err != nil {
-			return closeHandlers(fmt.Errorf("unable to create log file in archive: %w", err), lf)
+
+		name := strings.TrimPrefix(path, logPath)
```

Comment:
are endpoint logs in a flat dir? or is there any nesting in subdirs?

Reply:
as far as I know, on a flat dir. Why? if there are nested dir it wouldn't work?

Reply:
the names will get mangled on Windows if it's nested